### PR TITLE
[Archive Mode] 1. move serializablebytes and kvstorefortrie into protocol

### PR DESCRIPTION
**Pull request by CoderZhi**
prepare for follow up fix for archive mode
1. move serializabletypes and kvstoretire into protocol namespace
2. new kvstore with state reader

### CODE REVIEW

File: action/protocol/kvstorefortrie.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+func (kv *kvStoreForTrieWithStateReader) Start(context.Context) error {
+	return nil
+}
+
+func (kv *kvStoreForTrieWithStateReader) Stop(context.Context) error {
+	return nil
+}
+
+func (kv *kvStoreForTrieWithStateReader) Put(key []byte, value []byte) error {
+	return errors.New("not implemented")
```
Comment (dustinxie) [inline action/protocol/kvstorefortrie.go:85]: is it possible to (and should we) use panic?

File: action/protocol/kvstorefortrie.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+func (kv *kvStoreForTrieWithStateReader) Delete(key []byte) error {
+	return errors.New("not implemented")
```
Comment (dustinxie) [inline action/protocol/kvstorefortrie.go:89]: same here

**Review by dustinxie — APPROVED** [review]

**Review by Liuhaai — APPROVED** [review]

File: action/protocol/kvstorefortrie.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package protocol
```
Comment (dustinxie) [inline action/protocol/kvstorefortrie.go:7]: also, why need to move this file here?

File: action/protocol/kvstorefortrie.go (hunk 1 above)
Comment (CoderZhi) [reply action/protocol/kvstorefortrie.go:85]: error handling should be handled by logic component

File: action/protocol/kvstorefortrie.go (hunk 3 above)
Comment (CoderZhi) [reply action/protocol/kvstorefortrie.go:7]: you may check the follow up PRs

**Pull request merged**